lib/open_trip_planner_client/itinerary_group.ex

```diff
@@ -40,95 +41,91 @@ defmodule OpenTripPlannerClient.ItineraryGroup do
   def groups_from_itineraries(itineraries, opts \\ []) do
     itineraries
     |> Enum.group_by(&Itinerary.group_identifier/1)
-    |> Enum.map(&truncate_list(&1, opts))
-    |> Enum.reject(&Enum.empty?/1)
     |> Enum.map(&to_group(&1, opts))
-    |> Enum.sort_by(&tag_priority/1)
+    |> Enum.sort_by(&tag_and_cost_sorter/1)
     |> Enum.take(Keyword.get(opts, :num_groups, @num_groups))
   end
 
-  defp truncate_list({_, grouped_itineraries}, opts) do
+  defp truncate_list(grouped_itineraries, opts) do
     if opts[:take_from_end] do
       grouped_itineraries
-      |> ItineraryTag.sort_tagged(:end)
+      |> Enum.sort_by(&DateTime.to_unix(&1.end))
       |> Enum.take(-@max_per_group)
     else
       grouped_itineraries
-      |> ItineraryTag.sort_tagged(:start)
+      |> Enum.sort_by(&DateTime.to_unix(&1.start))
       |> Enum.take(@max_per_group)
     end
   end
 
-  defp to_group(grouped_itineraries, opts) do
-    representative_index = if(opts[:take_from_end], do: -1, else: 0)
+  defp to_group({_, all_itineraries}, opts) do
+    limited_itineraries = truncate_list(all_itineraries, opts)
+    summary = summary(all_itineraries)
+    representative_index = if(opts[:take_from_end], do: length(limited_itineraries) - 1, else: 0)
```

Comment:
While most of the code I had reading `representative_index` worked with the `-1` value, some didn't work. So setting this to the corresponding positive value seemed more robust for representing "the last itinerary".